app/src/main/java/com/andrew/tasky/agenda/data/AgendaApiRepositoryImpl.kt

```diff
@@ -10,23 +10,165 @@ import retrofit2.HttpException
 class AgendaApiRepositoryImpl(
     private val api: AgendaApi
 ) : AgendaApiRepository {
-    override suspend fun getAgendaItems(timezone: String, time: Long
+    override suspend fun getAgendaItems(
+        timezone: String,
+        time: Long
     ): AuthResult<List<AgendaItem>> {
-        TODO("Not yet implemented")
+        return try {
+            val response = api.getAgendaItems(timezone = timezone, time = time)
+            val tasks = response.tasks.map { task ->
+                AgendaItem(
+                    apiId = task.id,
+                    type = AgendaItemType.TASK,
```

Comment:
Didn't you use a sealed class for agenda items? Something like this:
```
sealed class AgendaItem(...) {
    data class Event(...): AgendaItem(...)
    data class Task(...): AgendaItem(...)
    data class Reminder(...): AgendaItem(...)
}
```

That way you don't need the type parameter since it's contained in the class already